### PR TITLE
4937 db query optimization

### PR DESCRIPTION
**Pull request by pkdash**
### Pull Request Checklist: 
- [x] Positive Test Case Written by Dev

- [x] Automated Testing

- [x] Sufficient User and Developer Documentation

- [x] Passing Jenkins Build

- [ ] Peer Code review and approval

### Positive Test Case
1. The db query related changes should improve the overall performance of hydroshare as it reduces the number of db queries in many of the request handlers. It may not be easy to verify the performance gain in manual testing.

### CODE REVIEW

**Comment by hydrocheck** [comment]
| METRIC | VALUE |
|:-----:|:-----:|

[https://sonarqube.cuahsi-workstation.com:9000/dashboard?id=hydroshare-4938](https://sonarqube.cuahsi-workstation.com:9000/dashboard?id=hydroshare-4938)

File: hs_core/tasks.py [hunk 1]
```diff
@@ -319,7 +321,7 @@ def send_over_quota_emails():
                                   [u.email, settings.DEFAULT_SUPPORT_EMAIL],
                                   html_message=msg_str)
                     except Exception as ex:
-                        logger.debug("Failed to send quota warning email: " + ex.message)
+                        logger.debug("Failed to send quota warning email: " + str(ex))
```
Comment (devincowan) [inline hs_core/tasks.py:324]: noticed a few more of these `ex.message` so I fixed them and pushed. Reject if you don't like! Hope that's ok

File: hs_core/views/__init__.py [hunk 2]
```diff
@@ -2451,7 +2452,7 @@ def dispatch(self, *args, **kwargs):
         return super(MyGroupsView, self).dispatch(*args, **kwargs)
 
     def get_context_data(self, **kwargs):
-        u = User.objects.get(pk=self.request.user.id)
+        u = User.objects.select_related("uaccess").get(pk=self.request.user.id)
```
Comment (devincowan) [inline hs_core/views/__init__.py:2455]: if we `select_related` here, does this impact line 2473 where we return `u`? I don't think so, but just checking...

File: hs_core/views/__init__.py [hunk 3]
```diff
@@ -2510,7 +2511,7 @@ def get_context_data(self, **kwargs):
 
         if self.request.user.is_authenticated:
             group_members = g.gaccess.members
-            u = User.objects.get(pk=self.request.user.id)
+            u = User.objects.select_related("uaccess").get(pk=self.request.user.id)
```
Comment (devincowan) [inline hs_core/views/__init__.py:2514]: if we `select_related` here, does that impact line 2555 `"profile_user": u,`

**Review by devincowan — APPROVED** [review]
looks good, a few questions...

File: hs_core/tasks.py (hunk 1 above)
Comment (pkdash) [reply hs_core/tasks.py:324]: Thanks for fixing this.

File: hs_core/views/__init__.py (hunk 2 above)
Comment (pkdash) [reply hs_core/views/__init__.py:2455]: Yes `uaccess` is accessible from `u` without hitting the db again.

File: hs_core/views/__init__.py (hunk 3 above)
Comment (pkdash) [reply hs_core/views/__init__.py:2514]: Same answer as above.

Comment (devincowan) [reply hs_core/views/__init__.py:2514]: thanks

**Review by devincowan — APPROVED** [review]

**Pull request merged**